libraries/chain/block_log.cpp

```diff
@@ -1133,50 +1136,44 @@ namespace eosio { namespace chain {
          fclose(ind_in);
    }
 
+   uint64_t trim_data::block_index(uint32_t n) const {
+      using namespace std;
+      EOS_ASSERT( n <= last_block, block_log_exception,
+                  "cannot seek in ${file} to block number ${b}, block number ${last} is the last block",
+                  ("file", index_file_name.string())("b",n)("last",last_block) );
+      return sizeof(uint64_t) * (n - first_block);
+   }
+
    uint64_t trim_data::block_pos(uint32_t n) {
       using namespace std;
-      index_pos = sizeof(uint64_t) * (n - first_block);
+      // can indicate the location of the block after the last block
+      if (n == last_block + 1) {
+         return ftell(blk_in);
+      }
+      const uint64_t index_pos = sizeof(uint64_t) * (n - first_block);
       auto status = fseek(ind_in, index_pos, SEEK_SET);
       EOS_ASSERT( status == 0, block_log_exception, "cannot seek to ${file} ${pos} from beginning of file for block ${b}", ("file", index_file_name.string())("pos", index_pos)("b",n) );
       const uint64_t pos = ftell(ind_in);
       EOS_ASSERT( pos == index_pos, block_log_exception, "cannot seek to ${file} entry for block ${b}", ("file", index_file_name.string())("b",n) );
       uint64_t block_n_pos;
       auto size = fread((void*)&block_n_pos, sizeof(block_n_pos), 1, ind_in);                   //filepos of block n
       EOS_ASSERT( size == 1, block_log_exception, "cannot read ${file} entry for block ${b}", ("file", index_file_name.string())("b",n) );
-      return block_n_pos;
-   }
 
-   void trim_data::find_block_pos(uint32_t n) {
-      //get file position of block n from blocks.index then confirm block n is found in blocks.log at that position
-      //sets fpos0 and fpos1, throws exception if block at fpos0 is not block n
-      using namespace std;
-      auto temp = block_pos(first_block);
-      fpos0 = block_pos(n);
-      auto size = fread((void*)&fpos1, sizeof(fpos1), 1, ind_in);                   //filepos of block n+1
-      if (n != last_block)
-         EOS_ASSERT( size == 1, block_log_exception, "cannot read ${file} entry for block ${b}, size=${size}",
-                     ("file", index_file_name.string())("b",n + 1)("size",size) );
-
-      cout << "According to blocks.index:\n";
-      cout << "    block " << n << " starts at position " << fpos0 << '\n';
-      cout << "    block " << n + 1;
-
-      if (n != last_block)
-         cout << " starts at position " << fpos1 << '\n';
-      else
-         cout << " is past end\n";
-
-      //read blocks.log and verify block number n is found at file position fpos0
-      auto status = fseek(blk_in, fpos0 + blknum_offset, SEEK_SET);
-      EOS_ASSERT( status == 0, block_log_exception, "cannot seek to ${file} ${pos} from beginning of file", ("file", block_file_name.string())("pos", fpos0 + blknum_offset) );
+      //read blocks.log and verify block number n is found at the determined file position
```

Comment:
Why is any of the rest of this necessary? Isn't it safe to assume that block.index hasn't been corrupted?

Reply:
I was just keeping the same logic as before.  If I remove it here, I need to add it to smoke_test in eosio-blocklog.  This may not be the most useful validation, but not sure that we really have a scenario where we can say that we have a reason to trust the index is correct.  Without this check the only validation we are doing is that the index position falls in a valid range in the block log file.  My vote is to leave this so that we are at least doing this and revisit index validation when we actually take a step back and think about what we want to be doing in this code.

Reply:
Sure, that's fine. We can keep it here.